mongodbatlas/resource_mongodbatlas_search_analyzer.go

```diff
@@ -0,0 +1,214 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/mwielbut/pointy"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+	"log"
+	"strings"
+)
+
+func resourceMongoDBAtlasSearchAnalyzers() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMongoDBAtlasSearchAnalyzersCreate,
+		Read:   resourceMongoDBAtlasSearchAnalyzersRead,
+		Update: resourceMongoDBAtlasSearchAnalyzersUpdate,
+		Importer: &schema.ResourceImporter{
+			State: resourceMongoDBAtlasSearchAnalyzersImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"search_analyzers": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"base_analyzer": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Required: false,
+						},
+						"ignore_case": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"max_token_length": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: false,
+						},
+						"stem_exclusion_set": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"stopwords": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasSearchAnalyzersImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	conn := meta.(*MongoDBClient).Atlas
+
+	parts := strings.SplitN(d.Id(), "-", 2)
+	if len(parts) != 2 {
+		return nil, errors.New("import format error: to import a search analyzer, use the format {project_id}-{cluster_name}")
+	}
+
+	projectID := parts[0]
+	clusterName := parts[1]
+
+	_, _, err := conn.Search.ListAnalyzers(context.Background(), projectID, clusterName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't import search analyzers in projectID (%s) and Cluster (%s), error: %s", projectID, clusterName, err)
+	}
+
+	if err := d.Set("project_id", projectID); err != nil {
+		log.Printf("[WARN] Error setting project_id for (%s): %s", projectID, err)
+	}
+
+	if err := d.Set("cluster_name", clusterName); err != nil {
+		log.Printf("[WARN] Error setting cluster_name for (%s): %s", clusterName, err)
+	}
+
+	d.SetId(encodeStateID(map[string]string{
```

Comment:
is this one per cluster?

Reply:
yes, because you only can set all the custom analyzers at the same time (there's no id for this custom analyzers)

https://docs.atlas.mongodb.com/reference/api/fts-analyzers-update-all/